collect_app/src/main/java/org/odk/collect/android/activities/FormFillingActivity.java

```diff
@@ -418,7 +423,14 @@ public void onCreate(Bundle savedInstanceState) {
                 .forClass(SelectOneFromMapDialogFragment.class, () -> new SelectOneFromMapDialogFragment(viewModelFactory))
                 .build());
 
+        savedInstanceState = savedInstanceStateProvider.getState(savedInstanceState);
```

Comment:
I don't feel great about this solution of forcing in a provider for `savedInstanceState`, but there's no existing way to start an Activity with previous state using `ActivityScenario`, and this is the best I could come up with (without forking AndroidX Test).

It currently **is** possible to write pure Robolectric code that that creates an Acivity with previous state so it should be possible to add that to `ActivityScenario` for "local" tests (that under the hood use the Robolectric backed `LocalActivityInvoker`), but I don't have enough knowledge of Android internals to know if it'd be doable for instrumentation tests. I'll file an issue over at AndroidX Test for this.

Reply:
Link to issue: https://github.com/android/android-test/issues/1825.